<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>pathfinding_simulation</class>
 <widget class="QWidget" name="pathfinding_simulation">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>552</width>
    <height>486</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="pathfinding_vertical_layout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="environment_group">
       <property name="title">
        <string>Scenario</string>
       </property>
       <layout class="QVBoxLayout" name="environment_vertical_layout">
        <item>
         <widget class="QRadioButton" name="saved_environment_radio">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>&amp;Load from file</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="random_generated_environment_radio">
          <property name="text">
           <string>&amp;Generate random scenarios</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="random_generated_options">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="scenario_size_x_label">
            <property name="text">
             <string>Scenario size (X):</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="scenario_size_x_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>1000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="scenario_size_y_label">
            <property name="text">
             <string>Scenario size (Y):</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="scenario_size_y_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>1000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="obstacle_number_label">
            <property name="text">
             <string>Number of obstacles:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="obstacle_number_input">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="min_obs_side_label">
            <property name="text">
             <string>Min. obstacle side:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="min_obs_side_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>5.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="max_obs_side_label">
            <property name="text">
             <string>Max. obstacle side:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="max_obs_side_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>40.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="min_obs_height_label">
            <property name="text">
             <string>Min. obstacle height:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QDoubleSpinBox" name="min_obs_height_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>4.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="max_obs_height_label">
            <property name="text">
             <string>Max. obstacle height:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QDoubleSpinBox" name="max_obs_height_input">
            <property name="suffix">
             <string> m</string>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>40.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="rotation_checkbox">
          <property name="text">
           <string>Rotate obstacles</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="environment_spacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>1</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="right_half_widget" native="true">
       <layout class="QVBoxLayout" name="right_half_vertical_layout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="simulations_group">
          <property name="title">
           <string>Simulations</string>
          </property>
          <layout class="QVBoxLayout" name="simulations_vertical_layout">
           <item>
            <widget class="QRadioButton" name="single_scenario_radio_button">
             <property name="text">
              <string>Simulate single scenario</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="monte_carlo_radio_button">
             <property name="text">
              <string>&amp;Monte Carlo simulation</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="monte_carlo_options_form">
             <item row="0" column="1">
              <widget class="QSpinBox" name="simulation_number_input">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="simulation_number_label">
               <property name="text">
                <string>Number of simulations:</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="simulations_spacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>0</width>
               <height>1</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="results_group">
          <property name="title">
           <string>Results</string>
          </property>
          <layout class="QVBoxLayout" name="results_vertical_layout">
           <item>
            <widget class="QRadioButton" name="full_results_radio_button">
             <property name="text">
              <string>Show full results</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="statistics_radio_button">
             <property name="text">
              <string>Show result statistics</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="results_spacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>0</width>
               <height>1</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="enabled_algorithms_group">
          <property name="title">
           <string>Enabled algorithms</string>
          </property>
          <layout class="QVBoxLayout" name="enabled_algorithms_vertical_layout">
           <item>
            <widget class="QCheckBox" name="reference_checkbox">
             <property name="text">
              <string>Use edge sampling algorithm</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="approximate_checkbox">
             <property name="text">
              <string>Use approximate algorithm</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="approximate_filtering_checkbox">
             <property name="text">
              <string>Use approximate algorithm with filtering</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="plane_cut_checkbox">
             <property name="text">
              <string>Use plane cut algorithm</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="prm_checkbox">
             <property name="text">
              <string>Use probabilistic roadmap</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="rrt_checkbox">
             <property name="text">
              <string>Use rapidly-exploring random trees</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="bitstar_checkbox">
             <property name="text">
              <string>Use Batch Informed Trees</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="algorithms_spacer_checkbox">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>0</width>
               <height>1</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="bottom_button_layout">
     <item>
      <spacer name="bottom_spacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>1</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="queue_simulation_button">
       <property name="text">
        <string>Queue simulation</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="start_simulation_button">
       <property name="text">
        <string>Start simulation</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <slots>
  <slot>adjust_limits()</slot>
 </slots>
</ui>
